include/RAJA/util/RuntimePluginLoader.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef RAJA_Runtime_Plugin_Loader_HPP
+#define RAJA_Runtime_Plugin_Loader_HPP
+
+#include "RAJA/util/PluginStrategy.hpp"
+#include "RAJA/util/PluginInit.hpp"
+
+#include <dlfcn.h>
+#include <dirent.h>
+#include <vector>
+#include <memory>
+
+using Plugin = RAJA::util::PluginStrategy;
+
+class PluginLoader : public Plugin
+{
+public:
+  PluginLoader()
+  {
+    char *env = ::getenv("RAJA_PLUGINS");
+    if (nullptr == env)
+    {
+      return;
+    }
+    loadDirectory(env);
+  }
+
+  void preLaunch(RAJA::util::PluginContext p)
+  {
+    // Checking for new plugin directories to add.
+    if (!RAJA::plugin::paths.empty())
```

Comment:
I think this code should be moved to the init, doing it every time preLaunch is called will add too much overhead.